clients/src/main/java/org/apache/kafka/common/telemetry/internals/ClientTelemetryUtils.java

```diff
@@ -203,17 +201,17 @@ public static byte[] compress(byte[] raw, CompressionType compressionType) throw
 
     public static ByteBuffer decompress(byte[] metrics, CompressionType compressionType) {
         ByteBuffer data = ByteBuffer.wrap(metrics);
-        try (InputStream in = compressionType.wrapForInput(data, RecordBatch.CURRENT_MAGIC_VALUE, BufferSupplier.create());
-            ByteArrayOutputStream out = new ByteArrayOutputStream()) {
-
+        try (InputStream in = compressionType.wrapForInput(data, RecordBatch.CURRENT_MAGIC_VALUE, BufferSupplier.create())) {
             byte[] bytes = new byte[data.capacity() * 2];
             int nRead;
+            int totalReads = 0;
             while ((nRead = in.read(bytes, 0, bytes.length)) != -1) {
-                out.write(bytes, 0, nRead);
+                totalReads+=nRead;
```

Comment:
you don't need to do this since `ByteBufferOutputStream` can expand inner buffer.